final-project/helpers.py

```diff
@@ -207,37 +207,37 @@ def get_reference(product_with_references):
 
     # Check whether we found references
     if not references_for_product:
-        return apology("We have no references. Please help us research")
+        return (None, None, None, None)
     elif references_for_product == None:
-        return apology("We have no references. Please help us research")
-
-    # Save the number of research articles for this products.
-    number_of_references = len(references_for_product)
-
-    # Save the titles and links to each reference.
-    # Create an empty list to save our ids.
-    reference_ids = []
-    # Create an empty list to save our titles.
-    reference_titles = []
-    # Create an empty list to save our links.
-    reference_links = []
-
-    # Loop through each title in our list of references
-    for reference_article in references_for_product:
-
-        # Save the id.
-        reference_ids.append(reference_article["id"])
-
-        # Ensure we have a title.
-        if not reference_article["title"]:
-            # If there is no title, then use the link.
-            reference_titles.append(reference_article["link"])
-        # If there is a title, then use the title.
-        else:
-            # Save the title.
-            reference_titles.append(reference_article["title"])
-
-        # Save the link.
-        reference_links.append(reference_article["link"])
-
-    return (number_of_references, reference_ids, reference_titles, reference_links)
+        return (None, None, None, None)
```

Comment:
Why not just empty return?

Reply:
I'll try that and test it. 

I got an (expected 4 values, received 2) from line 524 in my application.py file, but I think it might have been a fluke. When I restarted my server, the issue seemed to go away even with the empty return. I'll go back and test it again and see if there is not a better solution than the one I gave above.

Reply:
@hdavidzhu I fixed it. You were right, there was no need to return None. The error I received was because I needed to restart my server. 

This PR is still useful because it contains the UIs empty state when no references exist, so I will merge now.